### PR TITLE
refactor(v2): replaces more modals

### PR DESCRIPTION
**Pull request by dzucconi**
More modal deletion + updates a few to use the correct dialogs. Completely removes `ErrorModal`

### CODE REVIEW

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/TwoFactorAuthentication/Components/AppSecondFactor/index.tsx [hunk 1]
```diff
@@ -34,9 +41,23 @@ export const AppSecondFactor: React.FC<AppSecondFactorProps> = ({
     false
   )
   const [stagedSecondFactor, setStagedSecondFactor] = useState(null)
-  const [isDisabling] = useState(false)
+  const [isDisabling] = useState(false) // ???
   const [isCreating, setCreating] = useState(false)
 
+  // Reset's _all_ of these state flags.
+  // NOTE: Don't do this. Just hacking around the architecture here.
```
Comment (dzucconi) [inline src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/TwoFactorAuthentication/Components/AppSecondFactor/index.tsx:48]: The error handling in these components didn't work. Also just consider what the state machine for this component would look like if it was modeled... 😵

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/TwoFactorAuthentication/Components/AppSecondFactor/index.tsx [hunk 2]
```diff
@@ -34,9 +41,23 @@ export const AppSecondFactor: React.FC<AppSecondFactorProps> = ({
     false
   )
   const [stagedSecondFactor, setStagedSecondFactor] = useState(null)
-  const [isDisabling] = useState(false)
+  const [isDisabling] = useState(false) // ???
```
Comment (dzucconi) [inline src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/TwoFactorAuthentication/Components/AppSecondFactor/index.tsx:44]: 🔍

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/TwoFactorAuthentication/Components/AppSecondFactor/index.tsx [hunk 3]
```diff
@@ -91,12 +112,18 @@ export const AppSecondFactor: React.FC<AppSecondFactorProps> = ({
     }
   }
 
-  function handleMutationError(errors: ApiError[]) {
-    if (!Array.isArray(errors)) {
-      throw errors
-    }
+  function handleMutationError(err: ApiError[]) {
```
Comment (dzucconi) [inline src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/TwoFactorAuthentication/Components/AppSecondFactor/index.tsx:115]: This type is almost certainly wrong.

File: src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/TwoFactorAuthentication/Components/SmsSecondFactor/index.tsx [hunk 4]
```diff
@@ -163,18 +163,16 @@ export const SmsSecondFactor: React.FC<SmsSecondFactorProps> = ({
...
+          <Message
+            variant="warning"
+            title="Artsy employees are encouraged to use the “App Authenticator” 2FA
+          method via 1Password (or your preferred password manager)."
+          >
+            You may find a detailed walkthrough{" "}
+            <a href="https://artsy.net/employees-mfa-instructions">
+              here in Notion
+            </a>
+            .
```
Comment (dzucconi) [inline src/v2/Apps/Settings/Routes/EditSettings/Components/SettingsEditSettingsTwoFactor/TwoFactorAuthentication/Components/SmsSecondFactor/index.tsx:175]: cc @nickskalkin

File: src/v2/Apps/Artwork/Components/ArtworkSidebar/__tests__/ArtworkSidebarCommercial.jest.tsx [hunk 5]
```diff
@@ -44,14 +42,17 @@ describe("ArtworkSidebarCommercial", () => {
...
       <SystemContextProvider>
-        <ArtworkSidebarCommercialContainerWithInquiry
-          artworkID={artwork.internalID}
-          artwork={artwork}
-          user={user}
-          mediator={mediator}
-          relay={{ environment: {} } as RelayProp}
-          {...otherProps}
-        />
+        <ToastsProvider>
```
Comment (damassi) [inline src/v2/Apps/Artwork/Components/ArtworkSidebar/__tests__/ArtworkSidebarCommercial.jest.tsx:45]: Why is this necessary here?
